<a href="https://colab.research.google.com/github/Mathieu-Germain/LogicMojo-AI-ML-April26-Mathieu/blob/main/Practice_Maths_for_ML_practice_Notebook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

###Two dice are rolled. Find the probability that a) the sum is 7 b) both dice show the same number c) the product is even

### Probability Calculations for Two Dice Rolls

In [ ]:
from fractions import Fraction

# Define the sample space for rolling two dice
sample_space = [(d1, d2) for d1 in range(1, 7) for d2 in range(1, 7)]
total_outcomes = len(sample_space)

print(f"Total possible outcomes: {total_outcomes}")
# print(f"Sample space: {sample_space}") # Uncomment to see the full sample space

Total possible outcomes: 36


#### a) The sum is 7

In [ ]:
event_a_outcomes = [outcome for outcome in sample_space if sum(outcome) == 7]
probability_a = Fraction(len(event_a_outcomes), total_outcomes)
# print(f"Probability of a single event is: {probability_a}")
print(f"Outcomes where the sum is 7: {event_a_outcomes}")
print(f"Number of outcomes for event a: {len(event_a_outcomes)}")
print(f"Probability that the sum is 7: {probability_a} (approx. {probability_a:.4f})")

Probability of a single event is: 1/6
Outcomes where the sum is 7: [(1, 6), (2, 5), (3, 4), (4, 3), (5, 2), (6, 1)]
Number of outcomes for event a: 6
Probability that the sum is 7: 1/6 (approx. 0.1667)


#### b) Both dice show the same number

In [ ]:
event_b_outcomes = [outcome for outcome in sample_space if outcome[0] == outcome[1]]
probability_b = Fraction(len(event_b_outcomes), total_outcomes)

print(f"Outcomes where both dice show the same number: {event_b_outcomes}")
print(f"Number of outcomes for event b: {len(event_b_outcomes)}")
print(f"Probability that both dice show the same number: {probability_b} (approx. {probability_b:.4f})")

Outcomes where both dice show the same number: [(1, 1), (2, 2), (3, 3), (4, 4), (5, 5), (6, 6)]
Number of outcomes for event b: 6
Probability that both dice show the same number: 1/6 (approx. 0.1667)


#### c) The product is even

In [ ]:
# The product of two numbers is even if at least one of the numbers is even.
# This is equivalent to saying it's NOT (both numbers are odd).
# Outcomes where both are odd: (1,1), (1,3), (1,5), (3,1), (3,3), (3,5), (5,1), (5,3), (5,5)

event_c_outcomes = [outcome for outcome in sample_space if (outcome[0] * outcome[1]) % 2 == 0]
probability_c = Fraction(len(event_c_outcomes), total_outcomes)

print(f"Number of outcomes for event c (product is even): {len(event_c_outcomes)}")
print(f"Probability that the product is even: {probability_c} (approx. {probability_c:.4f})")

Number of outcomes for event c (product is even): 27
Probability that the product is even: 3/4 (approx. 0.7500)


Two dice are rolled. Find the probability that: a) the sum is 7 b) both dice show the same number c) the product is even
Sure, I can help you find those probabilities! I'll generate some Python code to simulate rolling two dice and calculate the probabilities for each scenario.

The probabilities have been calculated successfully!

## Here are the results:

a). The probability that the sum of the two dice is 7: 1/6 (approximately 0.1667) The outcomes for this event are: [(1, 6), (2, 5), (3, 4), (4, 3), (5, 2), (6, 1)]

b) The probability that both dice show the same number: 1/6 (approximately 0.1667) The outcomes for this event are: [(1, 1), (2, 2), (3, 3), (4, 4), (5, 5), (6, 6)]

c) The probability that the product of the two dice is even: 3/4 (approximately 0.7500) There are 27 outcomes where the product is even.

In [ ]:
A box contains 5 red and 3 blue balls. One ball is drawn at random. a) What is the probability it is red? b) If two balls are drawn (without replacement), what is the probability both are blue?

### Probability Calculations for Drawing Balls from a Box

In [ ]:
from fractions import Fraction

# Define the number of balls
num_red_balls = 5
num_blue_balls = 3
total_balls = num_red_balls + num_blue_balls

print(f"Total number of balls: {total_balls}")
print(f"Number of red balls: {num_red_balls}")
print(f"Number of blue balls: {num_blue_balls}")

Total number of balls: 8
Number of red balls: 5
Number of blue balls: 3


#### a) What is the probability it is red if one ball is drawn at random?

In [ ]:
probability_red = Fraction(num_red_balls, total_balls)

print(f"Probability of drawing one red ball: {probability_red} (approx. {probability_red:.4f})")

Probability of drawing one red ball: 5/8 (approx. 0.6250)


#### b) If two balls are drawn (without replacement), what is the probability both are blue?

In [ ]:
# Probability of drawing the first blue ball
prob_first_blue = Fraction(num_blue_balls, total_balls)

# After drawing one blue ball, there is one less blue ball and one less total ball
remaining_blue_balls = num_blue_balls - 1
remaining_total_balls = total_balls - 1

# Probability of drawing the second blue ball (given the first was blue)
prob_second_blue_given_first_blue = Fraction(remaining_blue_balls, remaining_total_balls)

# Probability of both being blue is the product of these two probabilities
probability_both_blue = prob_first_blue * prob_second_blue_given_first_blue

print(f"Probability of drawing the first blue ball: {prob_first_blue}")
print(f"Probability of drawing the second blue ball given the first was blue: {prob_second_blue_given_first_blue}")
print(f"Probability that both drawn balls are blue (without replacement): {probability_both_blue} (approx. {probability_both_blue:.4f})")

Probability of drawing the first blue ball: 3/8
Probability of drawing the second blue ball given the first was blue: 2/7
Probability that both drawn balls are blue (without replacement): 3/28 (approx. 0.1071)


### Generalized Probability Calculation Function

In [ ]:
from fractions import Fraction

def calculate_ball_drawing_probabilities(num_red_balls, num_blue_balls):
    """
    Calculates probabilities for drawing balls from a box.

    Args:
        num_red_balls (int): The number of red balls in the box.
        num_blue_balls (int): The number of blue balls in the box.

    Returns:
        tuple: A tuple containing:
            - Fraction: Probability of drawing one red ball.
            - Fraction: Probability of drawing two blue balls without replacement.
    """
    total_balls = num_red_balls + num_blue_balls

    if total_balls == 0:
        print("Error: No balls in the box.")
        return Fraction(0), Fraction(0)

    # a) Probability of drawing one red ball
    probability_red = Fraction(num_red_balls, total_balls)

    # b) Probability of drawing two blue balls (without replacement)
    if num_blue_balls < 2:
        probability_both_blue = Fraction(0)
    else:
        prob_first_blue = Fraction(num_blue_balls, total_balls)
        remaining_blue_balls = num_blue_balls - 1
        remaining_total_balls = total_balls - 1

        if remaining_total_balls == 0:
            probability_both_blue = Fraction(0)
        else:
            prob_second_blue_given_first_blue = Fraction(remaining_blue_balls, remaining_total_balls)
            probability_both_blue = prob_first_blue * prob_second_blue_given_first_blue

    return probability_red, probability_both_blue

In [ ]:
# Example usage: 7 red balls, 4 blue balls
example_num_red = 7
example_num_blue = 4

red_prob, both_blue_prob = calculate_ball_drawing_probabilities(example_num_red, example_num_blue)

print(f"For {example_num_red} red balls and {example_num_blue} blue balls:")
print(f"  a) Probability of drawing one red ball: {red_prob} (approx. {red_prob:.4f})")
print(f"  b) Probability of drawing two blue balls (without replacement): {both_blue_prob} (approx. {both_blue_prob:.4f})")

# Another example: 5 red balls, 3 blue balls (original problem)
print("\n--- Original problem verification ---")
original_red_prob, original_both_blue_prob = calculate_ball_drawing_probabilities(5, 3)
print(f"For 5 red balls and 3 blue balls:")
print(f"  a) Probability of drawing one red ball: {original_red_prob} (approx. {original_red_prob:.4f})")
print(f"  b) Probability of drawing two blue balls (without replacement): {original_both_blue_prob} (approx. {original_both_blue_prob:.4f})")

For 7 red balls and 4 blue balls:
  a) Probability of drawing one red ball: 7/11 (approx. 0.6364)
  b) Probability of drawing two blue balls (without replacement): 6/55 (approx. 0.1091)

--- Original problem verification ---
For 5 red balls and 3 blue balls:
  a) Probability of drawing one red ball: 5/8 (approx. 0.6250)
  b) Probability of drawing two blue balls (without replacement): 3/28 (approx. 0.1071)


#### Demonstrating the generalized function with an example

Okay, I see you have a new probability problem. I will generate the Python code to solve this for you, calculating the probabilities for drawing red and blue balls from the box.

The probabilities for drawing balls from the box have been calculated successfully!

Here are the results:

Total Balls: 8 (5 red, 3 blue)
a) The probability of drawing one red ball at random: 5/8 (approximately 0.6250)

b) The probability of drawing two blue balls without replacement: 3/28 (approximately 0.1071) (This is calculated as the probability of drawing the first blue ball (3/8) multiplied by the probability of drawing a second blue ball given the first was blue (2/7)).

A card is drawn from a deck. Find the probability that it is a) a heart, given that it is red b) a king, given that it is a face card

### Probability Calculations for Drawing Cards from a Deck

In [ ]:
from fractions import Fraction

# A standard deck has 52 cards.
# 4 suits: Hearts, Diamonds, Clubs, Spades
# 2 colors: Red (Hearts, Diamonds), Black (Clubs, Spades)
# Each suit has 13 cards: A, 2, 3, 4, 5, 6, 7, 8, 9, 10, J, Q, K

total_cards = 52

# Define relevant counts for the problem
num_red_cards = 26  # Hearts (13) + Diamonds (13)
num_hearts = 13
num_kings = 4       # One King per suit

# Face cards: Jack, Queen, King (3 per suit * 4 suits)
num_face_cards = 12 # J, Q, K of Hearts, Diamonds, Clubs, Spades

print(f"Total cards in a deck: {total_cards}")
print(f"Number of red cards: {num_red_cards}")
print(f"Number of hearts: {num_hearts}")
print(f"Number of face cards: {num_face_cards}")
print(f"Number of kings: {num_kings}")

Total cards in a deck: 52
Number of red cards: 26
Number of hearts: 13
Number of face cards: 12
Number of kings: 4


#### a) Probability that it is a heart, given that it is red P(Heart | Red)

In [ ]:
# The event is drawing a heart. The condition is that the card is red.
# The number of outcomes satisfying the condition (red cards) is 26.
# The number of outcomes satisfying both the event and the condition (red hearts) is 13.

# P(Heart | Red) = P(Heart and Red) / P(Red)
# P(Heart and Red) = Number of red hearts / Total cards = 13 / 52
# P(Red) = Number of red cards / Total cards = 26 / 52

# Alternatively, if we know it's red, our sample space reduces to the 26 red cards.
# Out of these 26 red cards, 13 are hearts.

probability_heart_given_red = Fraction(num_hearts, num_red_cards)

print(f"Number of red hearts: {num_hearts}")
print(f"Number of red cards: {num_red_cards}")
print(f"Probability it is a heart, given that it is red: {probability_heart_given_red} (approx. {probability_heart_given_red:.4f})")

Number of red hearts: 13
Number of red cards: 26
Probability it is a heart, given that it is red: 1/2 (approx. 0.5000)


#### b) Probability that it is a king, given that it is a face card P(King | Face Card)

In [ ]:
# The event is drawing a king. The condition is that the card is a face card.
# The number of outcomes satisfying the condition (face cards) is 12.
# The number of outcomes satisfying both the event and the condition (kings that are also face cards) is 4 (all kings are face cards).

# P(King | Face Card) = P(King and Face Card) / P(Face Card)
# P(King and Face Card) = Number of kings / Total cards = 4 / 52
# P(Face Card) = Number of face cards / Total cards = 12 / 52

# Alternatively, if we know it's a face card, our sample space reduces to the 12 face cards.
# Out of these 12 face cards, 4 are kings.

probability_king_given_face_card = Fraction(num_kings, num_face_cards)

print(f"Number of kings (which are also face cards): {num_kings}")
print(f"Number of face cards: {num_face_cards}")
print(f"Probability it is a king, given that it is a face card: {probability_king_given_face_card} (approx. {probability_king_given_face_card:.4f})")

Number of kings (which are also face cards): 4
Number of face cards: 12
Probability it is a king, given that it is a face card: 1/3 (approx. 0.3333)


##Two cards are drawn without replacement. Find the probability that a) the second card is a heart, given the first card is a heart b) the second card is a king, given the first card is a king c) draw 2 cards witout replacement, find prob of both being hearts

### Probability Calculations for Drawing Two Cards Without Replacement

In [ ]:
from fractions import Fraction

# Initial state of a standard deck
initial_total_cards = 52
initial_num_hearts = 13
initial_num_kings = 4

print(f"Initial total cards: {initial_total_cards}")
print(f"Initial number of hearts: {initial_num_hearts}")
print(f"Initial number of kings: {initial_num_kings}")

Initial total cards: 52
Initial number of hearts: 13
Initial number of kings: 4


#### a) The second card is a heart, given the first card is a heart

In [ ]:
# Event: First card drawn is a heart.
# State after first draw:
remaining_total_cards_a = initial_total_cards - 1 # 51 cards left
remaining_num_hearts_a = initial_num_hearts - 1   # 12 hearts left

probability_second_heart_given_first_heart = Fraction(remaining_num_hearts_a, remaining_total_cards_a)

print(f"Remaining hearts after first heart drawn: {remaining_num_hearts_a}")
print(f"Remaining total cards after first heart drawn: {remaining_total_cards_a}")
print(f"Probability that the second card is a heart, given the first is a heart: {probability_second_heart_given_first_heart} (approx. {probability_second_heart_given_first_heart:.4f})")

Remaining hearts after first heart drawn: 12
Remaining total cards after first heart drawn: 51
Probability that the second card is a heart, given the first is a heart: 4/17 (approx. 0.2353)


#### b) The second card is a king, given the first card is a king

In [ ]:
# Event: First card drawn is a king.
# State after first draw:
remaining_total_cards_b = initial_total_cards - 1 # 51 cards left
remaining_num_kings_b = initial_num_kings - 1     # 3 kings left

probability_second_king_given_first_king = Fraction(remaining_num_kings_b, remaining_total_cards_b)

print(f"Remaining kings after first king drawn: {remaining_num_kings_b}")
print(f"Remaining total cards after first king drawn: {remaining_total_cards_b}")
print(f"Probability that the second card is a king, given the first is a king: {probability_second_king_given_first_king} (approx. {probability_second_king_given_first_king:.4f})")

Remaining kings after first king drawn: 3
Remaining total cards after first king drawn: 51
Probability that the second card is a king, given the first is a king: 1/17 (approx. 0.0588)


#### c) Probability of drawing two hearts without replacement

In [ ]:
# Probability of the first card being a heart
prob_first_heart = Fraction(initial_num_hearts, initial_total_cards)

# Probability of the second card being a heart given the first was a heart (from part a)
# We can reuse the result from part a or recalculate it here for clarity:
# prob_second_heart_given_first_heart = Fraction(initial_num_hearts - 1, initial_total_cards - 1)

probability_both_hearts = prob_first_heart * probability_second_heart_given_first_heart

print(f"Probability of first card being a heart: {prob_first_heart}")
print(f"Probability of second card being a heart given first was heart: {probability_second_heart_given_first_heart}")
print(f"Probability of drawing two hearts without replacement: {probability_both_hearts} (approx. {probability_both_hearts:.4f})")

Probability of first card being a heart: 1/4
Probability of second card being a heart given first was heart: 4/17
Probability of drawing two hearts without replacement: 1/17 (approx. 0.0588)


###A factory has two machines, M1 and M2.
###M1 produces 60% of items, M2 produces 40%.
###Defective rates: 3% for M1, 5% for M2. If an item is found defective, what is the probability it was produced by M1?

To solve this problem, we use **Bayes' Theorem**, which allows us to find the probability of an event based on prior knowledge of conditions related to the event.

Here is the step-by-step breakdown:

### Step 1: Define the Events

* Let $M_1$ be the event that an item is produced by Machine 1.
* Let $M_2$ be the event that an item is produced by Machine 2.
* Let $D$ be the event that an item is defective.

### Step 2: Identify the Given Probabilities

* **Production Shares (Prior Probabilities):**
* $P(M_1) = 60\% = 0.60$
* $P(M_2) = 40\% = 0.40$


* **Defective Rates (Conditional Probabilities/Likelihoods):**
* Probability an item is defective *given* it came from $M_1$:  $P(D \mid M_1) = 3\% = 0.03$
* Probability an item is defective *given* it came from $M_2$:  $P(D \mid M_2) = 5\% = 0.05$



### Step 3: Find the Total Probability of a Defective Item, $P(D)$

To find the overall probability that any randomly selected item is defective, we use the **Law of Total Probability**. This counts defective items from both machines:


$$P(D) = [P(D \mid M_1) \times P(M_1)] + [P(D \mid M_2) \times P(M_2)]$$

Substitute the numbers into the formula:


$$P(D) = (0.03 \times 0.60) + (0.05 \times 0.40)$$

$$P(D) = 0.018 + 0.020$$

$$P(D) = 0.038$$

*(This means that 3.8% of all items produced in the factory are defective).*

---

### Step 4: Apply Bayes' Theorem

We want to find the conditional probability that the item came from Machine 1, *given* that we already know it is defective ($P(M_1 \mid D)$).

The formula for Bayes' Theorem is:


$$P(M_1 \mid D) = \frac{P(D \mid M_1) \times P(M_1)}{P(D)}$$

Substitute our calculated values into the formula:


$$P(M_1 \mid D) = \frac{0.03 \times 0.60}{0.038}$$

$$P(M_1 \mid D) = \frac{0.018}{0.038}$$

Now, simplify the fraction:


$$P(M_1 \mid D) = \frac{18}{38} = \frac{9}{19}$$

Convert to a decimal and percentage:


$$P(M_1 \mid D) \approx 0.4737 \text{ or } \mathbf{47.37\%}$$

### Final Answer

If an item is found to be defective, the probability that it was produced by Machine 1 is **$\frac{9}{19}$** (or approximately **$47.37\%$**).

#Exercise:
A certain disease affects 1% of the population. A test has:

Sensitivity (true positive rate) = 95%

Specificity (true negative rate) = 90% If a person tests positive, what is the probability they actually have the disease?

To solve this problem, we can use **Bayes' Theorem** or a simpler **hypothetical population method**. This problem is a classic example of how a relatively accurate medical test can yield counterintuitive results when the disease itself is rare (a phenomenon known as the **Base Rate Fallacy**).

Here are two different ways to understand and solve the problem:

---

### Method 1: The Hypothetical Population Method (Easiest to Visualize)

Let's imagine a population of **10,000 people**.

**Step 1: Divide the population into sick and healthy groups**

* **1% have the disease:** $10,000 \times 0.01 = \mathbf{100\text{ people are sick}}$.
* **The remaining 99% are healthy:** $10,000 - 100 = \mathbf{9,900\text{ people are healthy}}$.

**Step 2: Apply the test results to both groups**

* **Of the 100 sick people:** The test has a 95% sensitivity (true positive rate).
* $100 \times 0.95 = \mathbf{95\text{ True Positives}}$ (sick people who test positive).


* **Of the 9,900 healthy people:** The test has a 90% specificity (true negative rate). This means 90% test negative, but the remaining **10% will mistakenly test positive** (false positive rate).
* $9,900 \times 0.10 = \mathbf{990\text{ False Positives}}$ (healthy people who test positive).



**Step 3: Analyze the total pool of positive tests**
If you look at everyone who received a positive test result, your total pool is:


$$\text{Total Positive Tests} = 95 \text{ (True Positives)} + 990 \text{ (False Positives)} = \mathbf{1,085\text{ people}}$$

**Step 4: Find the final probability**
Out of those 1,085 people who tested positive, how many actually have the disease? Only the 95 true positives.


$$\text{Probability} = \frac{95}{1,085} \approx 0.08755 \text{ or } \mathbf{8.76\%}$$

---

### Method 2: Using Bayes' Theorem (Formal Probability)

#### 1. Define the Events

* $D$: The person has the disease.
* $H$: The person is healthy (does not have the disease).
* $Pos$: The person tests positive.

#### 2. Identify Given Values

* **Base Rate / Prior Probability:** $P(D) = 1\% = 0.01$
* **Complementary Probability:** $P(H) = 99\% = 0.99$
* **Sensitivity (True Positive Rate):** $P(Pos \mid D) = 95\% = 0.95$
* **Specificity (True Negative Rate):** $P(Neg \mid H) = 90\% = 0.90 \rightarrow$ Thus, the **False Positive Rate** is $P(Pos \mid H) = 10\% = 0.10$

#### 3. Find Total Probability of Testing Positive, $P(Pos)$

Using the Law of Total Probability:


$$P(Pos) = [P(Pos \mid D) \times P(D)] + [P(Pos \mid H) \times P(H)]$$

$$P(Pos) = (0.95 \times 0.01) + (0.10 \times 0.99)$$

$$P(Pos) = 0.0095 + 0.0990 = \mathbf{0.1085}$$

#### 4. Apply Bayes' Theorem

We want to find $P(D \mid Pos)$, the probability of having the disease given a positive test:


$$P(D \mid Pos) = \frac{P(Pos \mid D) \times P(D)}{P(Pos)}$$

$$P(D \mid Pos) = \frac{0.95 \times 0.01}{0.1085}$$

$$P(D \mid Pos) = \frac{0.0095}{0.1085} \approx \mathbf{0.0876}$$

---

### Final Answer

If a person tests positive, the probability that they actually have the disease is approximately **$8.76\%$** (or a fraction of $\frac{19}{217}$).

### Why is this so low?

Even though the test is highly accurate, the disease is very rare (1%). Because the healthy population is so large (99%), the small 10% error margin of the test generates **990 false positives**, which completely overwhelms the **95 true positives**. Therefore, most people who test positive in this scenario are actually completely healthy!

##Exercise:
There are three boxes:

Box A: 2 gold, 1 silver

Box B: 1 gold, 2 silver

Box C: 3 silver
A box is chosen at random and a coin (or ball) is drawn. If it is gold, what is the probability it came from Box A?

To solve this problem, we can use **Bayes' Theorem**, which updates the probability of an event (choosing Box A) based on new evidence (knowing the drawn coin is gold).

Here is the step-by-step breakdown:

### Step 1: Define the Events

* Let $A$ be the event that Box A is chosen.
* Let $B$ be the event that Box B is chosen.
* Let $C$ be the event that Box C is chosen.
* Let $G$ be the event that a gold coin is drawn.

### Step 2: Find the Prior Probabilities

Since a box is chosen completely at random out of three options, each box has an equal chance of being selected:

* $P(A) = \frac{1}{3}$
* $P(B) = \frac{1}{3}$
* $P(C) = \frac{1}{3}$

### Step 3: Find the Conditional Probabilities (Likelihoods)

Next, we calculate the probability of drawing a gold coin given that a specific box was chosen:

* **Box A** has 2 gold and 1 silver (3 total coins): $P(G \mid A) = \frac{2}{3}$
* **Box B** has 1 gold and 2 silver (3 total coins): $P(G \mid B) = \frac{1}{3}$
* **Box C** has 0 gold and 3 silver (3 total coins): $P(G \mid C) = \frac{0}{3} = 0$

### Step 4: Find the Total Probability of Drawing a Gold Coin, $P(G)$

Using the **Law of Total Probability**, we combine the probability of selecting each box with the probability of getting gold from that box:


$$P(G) = [P(G \mid A) \times P(A)] + [P(G \mid B) \times P(B)] + [P(G \mid C) \times P(C)]$$

Substitute the values:


$$P(G) = \left(\frac{2}{3} \times \frac{1}{3}\right) + \left(\frac{1}{3} \times \frac{1}{3}\right) + \left(0 \times \frac{1}{3}\right)$$

$$P(G) = \frac{2}{9} + \frac{1}{9} + 0 = \frac{3}{9} = \frac{1}{3}$$

So, the total probability of drawing a gold coin across the entire setup is $\frac{1}{3}$.

---

### Step 5: Apply Bayes' Theorem

We want to find the conditional probability that the box was Box A, *given* that the coin drawn is gold ($P(A \mid G)$).

The formula for Bayes' Theorem is:


$$P(A \mid G) = \frac{P(G \mid A) \times P(A)}{P(G)}$$

Substitute the values we calculated:


$$P(A \mid G) = \frac{\frac{2}{3} \times \frac{1}{3}}{\frac{1}{3}}$$

To simplify, since $\frac{1}{3}$ is in both the numerator and denominator, they cancel out:


$$P(A \mid G) = \frac{2}{3}$$

### Final Answer

If the drawn coin is gold, the probability it came from Box A is **$\frac{2}{3}$** (or approximately **$66.67\%$**).

Question 1:- What is the primary difference between a Z-test and a t-test?(Easy)
A) Z-test is used when the sample size is small; t-test is used for large samples.
B) Z-test is used when the population variance is known; t-test is used when the population variance is unknown.
C) Z-test is used for categorical data; t-test is used for continuous data.
D) Z-test is for paired samples; t-test is for independent samples.

### The correct answer is B) Z-test is used when the population variance is known; t-test is used when the population variance is unknown.

Detailed Explanation:
* *Z-test: You use a Z-test when  - you know the true population standard deviation ($\sigma$)—and by extension, the population variance ($\sigma^2$). Because the exact variance is known, the test statistic follows a standard normal distribution (Z-distribution). It is also typically applied when the sample size is large ($n \ge 30$), which justifies the assumption of normality via the Central Limit Theorem.

* *t-test: You use a t-test when the population variance is unknown. Instead, you have to estimate it using the sample standard deviation ($s$). Because estimating the variance from a sample introduces extra uncertainty, the test statistic follows a t-distribution. The t-distribution has heavier, fatter tails to account for this added variability, which is especially critical when working with small sample sizes ($n < 30$).

Why the other options are incorrect:

* A is reversed: Z-tests are typically used for large samples ($n \ge 30$), whereas t-tests are historically designed for small samples (though a t-test can also be used effectively on larger sample sizes).
* C is incorrect: Both Z-tests and t-tests are parametric tests designed for continuous data where you are comparing means (e.g., heights, test scores, revenue). Categorical data would typically require tests like Chi-Square.
* D is incorrect: Both Z-tests and t-tests have variations that can handle both paired (dependent) samples and independent samples.

* Question 2:
- What is the purpose of ANOVA (Analysis of Variance)?(Easy)

A) To test the difference between the means of two groups.

B) To test the difference between the means of three or more groups.

C) To determine the relationship between two variables.

D) To predict a dependent variable from an independent variable.

##The correct answer is B) To test the difference between the means of three or more groups.

##Why this is the case:
* ANOVA (Analysis of Variance): While the name focuses on "variance," its primary purpose is to compare the means of three or more independent groups to see if at least one of them is significantly different from the others.

* It accomplishes this by analyzing the variance between the groups relative to the variance within the groups.

* If you tried to use multiple t-tests to compare three or more groups (e.g., Group A vs. B, B vs. C, and A vs. C), you would drastically increase the risk of making a Type I error (a false positive). ANOVA solves this by testing all the groups simultaneously.

##Quick breakdown of the other options:
* A refers to a t-test: A two-sample t-test (or Z-test) is specifically used when you only have two groups to compare.
* C refers to Correlation: Determining the strength and direction of a relationship between two variables is the specific job of correlation analysis (like Pearson's $r$).
* D refers to Regression: Predicting a dependent variable based on one or more independent variables is the definition of linear or multiple regression analysis.

Question 3:

Which of the following is a correct interpretation of a p-value in hypothesis testing?(Easy)
  * A) The probability that the null hypothesis is true.
  * B) The probability of observing the data given that the null hypothesis is true.
  * C) The probability of making a Type I error.
  * D) The significance level of the test.

## The correct answer is B) The probability of observing the data given that the null hypothesis is true.

# Why this is the case:
  * The definition of a p-value: The p-value measures the probability of obtaining test results at least as extreme as the results actually observed, under the assumption that the null hypothesis ($H_0$) is correct.
  
  * It answers the question: "If the baseline assumption (the null) is perfectly true, how unusual or surprising is my sample data?".A very small p-value means the data would be highly unlikely under the null hypothesis, leading you to reject it.
  
  # Quick Breakdown of the other options:A is a common misconception:
  
  * A p-value is not the probability that the null hypothesis is true. In classical (frequentist) statistics, a hypothesis is either true or false; it does not have a probability assigned to it.
  
  * C refers to the Alpha ($\alpha$) level set by the researcher: The probability of committing a Type I error (falsely rejecting a true null hypothesis) is exactly equal to your pre-determined significance level ($\alpha$), usually set to 0.05 or 5%.
  *  D is incorrect: The significance level ($\alpha$) is a static threshold chosen before running the test (e.g., $\alpha = 0.05$). The p-value is an output calculated from the collected data. You compare the p-value to the significance level to make a decision.

###Question 4:

In hypothesis testing, what does a Type II error represent?(Easy)

A) Rejecting the null hypothesis when it is true.

B) Accepting the null hypothesis when it is false.

C) Rejecting the alternative hypothesis when it is true.

D) Accepting the alternative hypothesis when it is false.


# The correct answer is B) Accepting the null hypothesis when it is false.


(Note: In formal statistical terms, we say "failing to reject" the null hypothesis rather than "accepting" it, but the core concept remains the same.)

##Why this is the case:
A Type II error occurs when there is a real effect or difference in the population (meaning the null hypothesis is actually false), but your test fails to detect it. Think of it as a false negative.

##Quick Breakdown of the other options:

* A represents a Type I error: Rejecting the null hypothesis when it is actually true is a false positive (discovering an "effect" that doesn't really exist).


* C and D are incorrect phrasing: Hypothesis testing is structurally set up to frame decisions strictly around the null hypothesis (either rejecting it or failing to reject it), rather than formally "accepting or rejecting the alternative".

###Question 5:-

You conduct a two-sample t-test and obtain a p-value of 0.03. If the significance level (α) is 0.05, what can you conclude?(Medium)
A) There is not enough evidence to reject the null hypothesis.
B) There is enough evidence to reject the null hypothesis.
C) The test should be rerun with a larger sample size.
D) The test results are inconclusive.

The correct answer is B) There is enough evidence to reject the null hypothesis.

##Why this is the case:
In hypothesis testing, you make your final statistical decision by comparing your calculated p-value directly against your pre-determined significance level ($\alpha$):

* The Decision Rule: If the p-value $\le \alpha$, you reject the null hypothesis ($H_0$). This indicates that the observed sample data is statistically significant and highly unlikely to have occurred purely due to random chance.

* Your Numbers: Your calculated p-value is 0.03 and your significance level ($\alpha$) is 0.05. Because 0.03 $\le$ 0.05, the p-value successfully clears the threshold required for statistical significance. As a result, you conclude that there is sufficient statistical evidence to reject the null hypothesis.

##Quick Breakdown of the other options:
* A is incorrect because concluding there is not enough evidence to reject the null hypothesis would require a p-value greater than your significance level (for example, if your p-value was 0.08).
* C is incorrect because you already achieved a statistically significant result with your current sample size, making a larger sample size unnecessary to establish a finding.
* D is incorrect because applying a strict binary decision rule (p-value $\le \alpha$) ensures that frequentist hypothesis testing always leads to a definitive conclusion—either rejecting or failing to reject the null hypothesis.

Question 6:-

In a clinical trial, you want to compare the mean blood pressure of patients under two different treatments. The data is paired since each patient receives both treatments. Which test would you use, and why?(Medium)

A) Independent samples t-test because the samples are different.

B) Paired samples t-test because the data is paired.

C) One-way ANOVA because there are two treatments.

D) Z-test because the population standard deviation is known.

The correct answer is **B) Paired samples t-test because the data is paired.**

### Why this is the case:

*
**Paired Data:** The question explicitly states that the data is paired. This means that each measurement under Treatment A corresponds directly to a measurement under Treatment B for the same subject (or a closely matched pair). A paired samples t-test analyzes the differences between these paired observations.


*
**Comparing Two Means:** You are comparing the mean blood pressure of two different treatments. When comparing exactly two means and the population standard deviation is not provided, a t-test is the appropriate statistical tool.



---

### Quick Breakdown of the other options:

*
**A is incorrect:** Choosing an independent samples t-test contradicts the prompt, which explicitly states that the data is paired rather than independent. Using an independent t-test on paired data would ignore the correlation between the pairs and reduce the statistical power of the test.


*
**C is incorrect:** While you technically can use ANOVA for two groups, it is specifically designed for comparing three or more independent groups. Furthermore, a standard one-way ANOVA assumes independent samples, not paired data.


*
**D is incorrect:** The prompt does not state or imply that the true population standard deviation ($\sigma$) is known. Without having this population parameter, a Z-test cannot be performed.

##Question 7:-

A researcher is using a One-Way ANOVA to compare the means of four groups. The ANOVA output shows a significant F-statistic. What should the researcher do next?(Medium)

A) Conclude that all group means are equal.

B) Perform a post hoc test to identify which specific group means are different.

C) Increase the sample size and rerun the ANOVA.

D) Conduct a paired t-test on all pairs of groups.


The correct answer is **B) Perform a post hoc test to identify which specific group means are different.**

### Why this is the case:

*
**The Omnibus Test:** An ANOVA is considered an "omnibus" test. When it yields a significant F-statistic, it tells you that at least one group mean is significantly different from the others. However, it does not tell you which specific groups differ.


*
**Post Hoc Tests:** To pinpoint the exact pairs that have significant differences (e.g., is Group 1 different from Group 2, or Group 1 from Group 4?), you must run follow-up tests called post hoc tests (such as Tukey's HSD or Bonferroni adjustments). These tests are specially designed to control the overall Type I error rate when making multiple comparisons.



---

### Quick Breakdown of the other options:

*
**A is completely opposite:** A significant F-statistic means you reject the null hypothesis that all group means are equal.


*
**C is unnecessary:** You already have a statistically significant result. Rerunning the test with a larger sample size is a waste of resources and doesn't solve the problem of finding where the differences lie.


*
**D is dangerous:** Conducting standard paired (or independent) t-tests on all possible pairs without adjustments would rapidly increase your family-wise error rate, leading to a high risk of false positives (Type I errors). Post hoc tests handle multiple comparisons safely.

Question 8:-
A researcher conducts a one-tailed z-test at a 1% significance level. If the critical z-value is 2.33, what would be the decision if the calculated z-value is 2.1?(Hard)

A) Reject the null hypothesis.

B) Fail to reject the null hypothesis.

C) Accept the null hypothesis.

D) Cannot be determined from the information given.

###The correct answer is **B) Fail to reject the null hypothesis.**

### Why this is the case:

To make a decision in hypothesis testing using the critical value approach, you must compare your calculated test statistic (the sample's actual z-value) against the given critical boundary.

*
**The Decision Rule:** For a one-tailed (right-tailed) test, you reject the null hypothesis if your calculated z-value is greater than or equal to your critical value of 2.33.


* **Your Numbers:** Your calculated z-value is 2.1, which is less than the critical value of 2.33. Because your test statistic does not cross the threshold into the rejection region, the results are not considered statistically significant.
* **Conclusion:** Therefore, you fail to reject the null hypothesis.

---

### Quick Breakdown of the other options:

* **A is incorrect:** Rejecting the null hypothesis would require your calculated z-value to be 2.33 or higher.

* **C is a common misconception:** In frequentist statistics, we never formally "accept" the null hypothesis ; we only ever "fail to reject" it. You cannot prove the null hypothesis is true; you can only conclude that you don't have enough evidence to prove it false.


* **D is incorrect:** Because you were provided with both the calculated z-value (2.1) and the critical threshold (2.33), you have all the mathematical information needed to make a definitive statistical decision.

A factory has a machine that dispenses 80 mL of fluid in a bottle. An employee believes the average amount of fluid is not 80 mL. Using 40 samples, he measures the average amount dispensed by the machine to be 78 mL with a standard deviation of 2.5.
(a) State the null and alternative hypotheses.
(b) At a 95% confidence level, is there enough evidence to support the idea that the machine is not working properly?

Here is the step-by-step solution to your statistical problem:

### (a) State the null and alternative hypotheses

The null hypothesis ($H_0$) represents the status quo or the baseline assumption that the machine is working perfectly as intended (dispensing exactly 80 mL). The alternative hypothesis ($H_1$) represents the employee's claim that the average amount is not 80 mL, which implies a two-tailed test since it could be more or less.

*
**Null Hypothesis ($H_0$):** $\mu=80$ (The population mean volume dispensed is 80 mL).


*
**Alternative Hypothesis ($H_1$):** $\mu\neq80$ (The population mean volume dispensed is not 80 mL).



---

### (b) At a 95% confidence level, is there enough evidence to support the idea that the machine is not working properly?

**Step 1: Choose the Test & Set up the Criteria**

* Because the sample size is large ($n=40$, which is $\ge30$), we can use the standard Normal distribution or Z-test approach.


* For a two-tailed test at a 95% confidence level (which means a significance level of $\alpha=0.05$), the critical Z-values are $\pm1.96$.


*
**Rejection Rule:** We will reject the null hypothesis if our calculated $Z\ge1.96$ or $Z\le-1.96$.



**Step 2: Calculate the Test Statistic ($Z$)**
We can set up the formula for the test statistic ($Z$):


$$Z = \frac{\bar{x} - \mu}{\frac{s}{\sqrt{n}}}$$

Plugging in the known values from your problem ($\bar{x}=78$, $\mu=80$, $n=40$, $s=2.5$):

1. First, calculate the standard error ($\frac{s}{\sqrt{n}}$):

$$\text{Standard Error} = \frac{2.5}{\sqrt{40}} \approx \frac{2.5}{6.325} \approx 0.395$$


2. Now, solve for $Z$:

$$Z = \frac{78 - 80}{0.395} = \frac{-2}{0.395} \approx -5.06$$



**Step 3: Compare and Make a Decision**

* Our calculated Z-value is **-5.06**.
* Because -5.06 is much smaller than the critical boundary of -1.96, it falls far into the rejection region.

**Conclusion:**
Yes, you **reject the null hypothesis**. There is enough evidence at the 95% confidence level to support the idea that the machine is not working properly and is malfunctioning. The observed sample data is statistically significant and highly unlikely to have occurred purely due to random chance.

Avg weight of residents is 168 kg. A nutrinitionist belives the mean is different from this.

She measures the wt of 36 individuals and found teh mean to be 169.5 kgs with a S.D of 3.9
is there enough evidence to discard the null hypothesis

Based on the statistical principles from the provided coursework, here is the step-by-step hypothesis test to determine if there is enough evidence to discard the null hypothesis.

### Step 1: State the Hypotheses

The nutritionist wants to know if the mean weight is different from 168 kg, which indicates a two-tailed test.

*
**Null Hypothesis ($H_0$):** $\mu = 168 \text{ kg}$ (The average weight of residents is 168 kg).


*
**Alternative Hypothesis ($H_1$):** $\mu \neq 168 \text{ kg}$ (The average weight of residents is different from 168 kg).



### Step 2: Choose the Test & Set up the Criteria

* **Sample size ($n$):** $36$. Since $n \ge 30$, we can safely use the standard Normal distribution or Z-test approach.


*
**Sample Mean ($\bar{x}$):** $169.5 \text{ kg}$.


*
**Sample Standard Deviation ($s$):** $3.9 \text{ kg}$.



(Note: Since a significance level $\alpha$ wasn't explicitly stated, we will default to the standard 95% confidence level, meaning $\alpha = 0.05$. For a two-tailed Z-test at 95% confidence, the critical Z-values are $\pm 1.96$.)

### Step 3: Calculate the Test Statistic ($Z$)

Using the test statistic formula:


$$Z = \frac{\bar{x} - \mu}{\frac{s}{\sqrt{n}}}$$

1. First, calculate the standard error ($\frac{s}{\sqrt{n}}$):


$$\text{Standard Error} = \frac{3.9}{\sqrt{36}} = \frac{3.9}{6} = 0.65$$





2. Now, solve for $Z$:


$$Z = \frac{169.5 - 168}{0.65} = \frac{1.5}{0.65} \approx 2.31$$






Our calculated test statistic is **$Z = 2.31$**.

### Step 4: Compare and Make a Decision

*
**Rejection Rule:** Reject $H_0$ if our calculated $Z \ge 1.96$ or $Z \le -1.96$.


*
**Your Value:** $2.31$ is greater than $1.96$ ($2.31 > 1.96$).
(Alternatively, checking a standard normal distribution table for $Z = 2.31$ gives a two-tailed p-value of approximately 0.0208. Since $0.0208 \le 0.05$, the result is statistically significant.)



### Conclusion

**Yes, there is enough evidence to discard (reject) the null hypothesis**. The sample mean weight of 169.5 kg is far enough away from 168 kg to be considered statistically significant at the 95% confidence level. You can confidently conclude that the true mean weight of the residents is significantly different from 168 kg.

A factory manufactures batteries with avg life span of 2 or more years. An engineer believes this value to be less.

Using 10 samples he measures the avg life span to be 1.8 years with a S.D of 0.15.
At 99% confidence is there enough evidence to discard the null hypothesis

To determine if there is enough evidence to discard (reject) the null hypothesis, let's go through the steps of a hypothesis test based on the information provided.

### Step 1: State the Hypotheses

The factory states that the average lifespan is 2 or more years ($\ge 2$). The engineer believes it is less ($< 2$), making this a left-tailed (one-tailed) test.

*
**Null Hypothesis ($H_0$):** $\mu \ge 2$ (The average battery lifespan is at least 2 years).


*
**Alternative Hypothesis ($H_1$):** $\mu < 2$ (The average battery lifespan is less than 2 years).



---

### Step 2: Identify the Test & Rejection Region

* **Sample size ($n$):** 10
* Since the sample size is small ($n < 30$) and the population standard deviation ($\sigma$) is not provided, we must use a **one-sample t-test**.


*
**Degrees of Freedom ($df$):** $n - 1 = 10 - 1 = 9$.


*
**Significance Level ($\alpha$):** 0.01 (at 99% confidence).



Using a standard t-distribution table for a left-tailed test with $df = 9$ and $\alpha = 0.01$, the critical value is $t = -2.821$.

*
**Rejection Rule:** We will reject the null hypothesis only if our calculated test statistic is more extreme than our boundary ($t \le -2.821$).



---

### Step 3: Calculate the Test Statistic ($t$)

To find out where the engineer's sample lands, we use the t-test formula:


$$t = \frac{\bar{x} - \mu}{\frac{s}{\sqrt{n}}}$$

Using your provided measurements ($\bar{x} = 1.8$, $s = 0.15$, $n = 10$):

1. **Calculate the standard error:**

$$\text{Standard Error} = \frac{0.15}{\sqrt{10}} \approx \frac{0.15}{3.162} \approx 0.0474$$


2. **Calculate the t-statistic:**

$$t = \frac{1.8 - 2}{0.0474} = \frac{-0.2}{0.0474} \approx -4.219$$



Our calculated test statistic is **$t \approx -4.22$**.

---

### Step 4: Compare and Make a Decision

* If your calculated t-value $\le -2.821$, you reject the null hypothesis.


* Our calculated value of **-4.22** is much smaller than **-2.821**, meaning it falls deep into the rejection region.

**Conclusion:**
Yes, you **reject the null hypothesis**. There is enough evidence at the 99% confidence level to support the engineer's belief that the battery lifespan is less than 2 years.

A factory manufactures cars with 5 year warranty. An engineer believes that the engine malfunctions in less than 5 years.


He takes a sample fo 20 cars and finds the avg to be 4.8 and a S.D of 0.50. At 2% significance can we reject the null hypothesis


To determine if there is enough evidence to reject the null hypothesis, let's go through the steps of a hypothesis test based on the information provided.

### Step 1: State the Hypotheses

The factory claims the warranty (and thus the average lifespan before malfunction) is 5 years. The engineer believes it is less than 5 years ($< 5$), making this a left-tailed (one-tailed) test.

* **Null Hypothesis ($H_0$):** $\mu \ge 5$ (The average engine malfunction time is at least 5 years).
* **Alternative Hypothesis ($H_1$):** $\mu < 5$ (The average engine malfunction time is less than 5 years).

---

### Step 2: Identify the Test & Rejection Region

* **Sample size ($n$):** 20
* Since the sample size is small ($n < 30$) and the population standard deviation ($\sigma$) is not provided, we must use a **one-sample t-test**.
* **Degrees of Freedom ($df$):** $n - 1 = 20 - 1 = 19$.
* **Significance Level ($\alpha$):** 0.02 (2%).

Using a standard t-distribution table (or calculator) for a left-tailed test with $df = 19$ and $\alpha = 0.02$, the critical value is **$t \approx -2.205$**.

* **Rejection Rule:** We will reject the null hypothesis only if our calculated test statistic is more extreme than our boundary ($t \le -2.205$).

---

### Step 3: Calculate the Test Statistic ($t$)

We use the t-test formula:


$$t = \frac{\bar{x} - \mu}{\frac{s}{\sqrt{n}}}$$

Using your provided measurements ($\bar{x} = 4.8$, $\mu = 5$, $s = 0.50$, $n = 20$):

1. **Calculate the standard error:**

$$\text{Standard Error} = \frac{0.50}{\sqrt{20}} \approx \frac{0.50}{4.472} \approx 0.1118$$


2. **Calculate the t-statistic:**

$$t = \frac{4.8 - 5}{0.1118} = \frac{-0.2}{0.1118} \approx -1.789$$



Our calculated test statistic is **$t \approx -1.789$**.

---

### Step 4: Compare and Make a Decision

* If your calculated t-value $\le -2.205$, you reject the null hypothesis.
* Our calculated value of **-1.789** is greater than **-2.205**. It does not cross the critical threshold into the rejection region. *(Additionally, the p-value for this result is approximately 0.0448, which is greater than the 0.02 significance level).*

**Conclusion:**
**No, we fail to reject the null hypothesis.** At the 2% significance level, there is not enough statistical evidence to support the engineer's belief that the engines malfunction in less than 5 years.